File: 2717. Semi-Ordered Permutation/Solution.cpp
class Solution {
public:
    int semiOrderedPermutation(vector<int>& nums) {
        int n = nums.size(), oneIdx = -1, nIdx = -1;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 1) {
                oneIdx = i;
            } else if (nums[i] == n) {
                nIdx = i;
            }
        }
        return (oneIdx < nIdx) ? oneIdx + (n - 1 - nIdx) : oneIdx + (n - 1 - nIdx) - 1;
    }
};

File: 2717. Semi-Ordered Permutation/Solution.py
class Solution:
    def semiOrderedPermutation(self, nums: List[int]) -> int:
        n = len(nums)
        oneIdx, nIdx = -1, -1
        for i in range(n):
            if nums[i] == 1:
                oneIdx = i
            elif nums[i] == n:
                nIdx = i
        return oneIdx + (n - 1 - nIdx) if oneIdx < nIdx else oneIdx + (n - 1 - nIdx) - 1
